### ISSUE TITLE
Bind IP

### ISSUE THREAD

**Issue by Neolo** [catalinii/minisatip#836]
Why there is no elementary option to specify BIND IP? Tvheadend is unable to detect minisatip running on other port but 554. And I want to run 2 instances of minisatip and surprisingly I can't find a way to bind a second copy to a specific local IP/adapter. Why? It's just all goes to 0.0.0.0:* no matter what.

**Comment by catalinii** [comment]
As far as I know you should be able to get tvh to work if you specify other ports instead of the default (-x and -y options)
the default ssdp announcement allows specifying the port as well

**Comment by Neolo** [comment]
> As far as I know you should be able to get tvh to work if you specify other ports instead of the default (-x and -y options)
> the default ssdp announcement allows specifying the port as well

None of those are valid flags for TVH, they are not described in help. And nothing I can find about how to force TVH to look for minisatip on non-default port.

**Comment by Neolo** [comment]
I'm ending up using one minisatip instance (which is sux, because if one tuner needs a restart, I have to restart entire service, taking down all tuners).
Because of this issue https://github.com/catalinii/minisatip/issues/837 I can't direct TVH over arguments like: 
TVH_ARGS="--satip_xml http://localhost:8081/desc.xml --satip_xml http://localhost:8082/desc.xml" etc, because they end up having same IDs and one overrides another.

**Comment by lars18th** [comment]
Hi @Neolo ,

I feel you can solve your problem if you try to not use the autodiscovering part of the SAT>IP standard. Even if this SSDP announcements and XML description files are useful for user deployment, they don't help in complex environments. Specifically inside an enviroment where multiple SAT>IP servers use the same IP. This scenerio is not supported directly by the standard, as it dictates one IP by device.

However, you can use several SAT>IP servers in the same device (IP address). You only need to define for each one:
- The RTSP port: 554 by default
- The range of pairs RTP ports: Any value, but it's preferable to not overlap.

Then you can execute the SAT>IP servers without the announcement enabled (in minisatip you can use `-A`). And for each client that requires to use the server, use the "manual discovering" (not specified in the standard). That's to set the IP:PORT, where the port is the RTSP listening port. In the TVH you can configure manually every SAT>IP server using the IP and PORT. And remember that in the _client_ if you want to use multiple SAT>IP servers, you need to reserve a non-overlaped RTP range ports. So reserver different ranges for different server. This is true even if the SAT>IP servers use different IPs.

Using this technique you can execute multiple SAT>IP instances inside the same server without troubles.
Regards.

**Comment by Neolo** [comment]
> And for each client that requires to use the server, use the "manual discovering"

Hi, thank you. In my case, there is 2 instances of minisatip and one client - Tvheadend (TVH), which suppose to be connected to both minisatips at a time. TVH due to it's poor development (well, sorry but it's true and goes over many years), there is no other way to add sat>ip servers manually, except for supplying .xml URL of those servers. I'm not sue, -A option will disable xml file generation as well?
The problem is that TVH sees only one minisatip instance at a time, and I guess it's because they use the same ip.
![Screen Shot 2021-04-20 at 10 15 19 PM](https://user-images.githubusercontent.com/2836790/115520481-ea1a1680-a225-11eb-8355-5bb2f0423818.png)

**Comment by lars18th** [comment]
Hi @Neolo ,

The TVH is not a bad project. However, it's development is very slow at time.

If the only way configuring a SAT>IP client is with the `--satip_xml` command-line parameter, then use the `-X` parameter with minisatip to link your homemade XML file. You can start the minisatip and download the default autogerated XML file with a browser. Then copy the file to the HTML subdirectory and name it `desc-1.xml` and copy to `desc-2.xml` too. Edit these two files as you need, and start minisatip using the `-X /opt/minisatip/html/desc-1.xml`.

I hope this helps you.

**Comment by lars18th** [comment]
Hi @Neolo ,

Two additional comments:
1. The correct use is `-X desc-1.xml` (or  `-X desc-2.xml`).
2. And even you use files "desc-1.xml" and "desc-2.xml", the path for the HTTP request continues to be "http://server:port/desc.xml".

So to start your TVheadend you need to use: `--satip_xml http://localhost:8081/desc.xml --satip_xml http://localhost:8082/desc.xml`.

Regards.

**Comment by lars18th** [comment]
Hi @Neolo ,

I suggest to change the TITLE of this issue to something more descriptive. For example: "How to run two instances of minisatip in the same server (sharing IP address)".

**Comment by Neolo** [comment]
> Hi @Neolo ,
> 
> I suggest to change the TITLE of this issue to something more descriptive. For example: "How to run two instances of minisatip in the same server (sharing IP address)".

Why is that? Minisatip does not have an option to bind to a specific IP on the host, _period_.
Whatever else discussed after are simply consequences and examples.

**Comment by 9000h** [comment]
hmm, why should minisatip fix tvh issues

**Comment by Neolo** [comment]
> hmm, why should minisatip fix tvh issues

Don't shift it to TVH. Bind-IP is an elementary option in any server software. Give me one reason why it wasn't done here.

**Comment by lars18th** [comment]
Hi,

I think it's a valid request to add support to Bind to an specific IP for minisatip. In fact, I've requested this years ago. But, at time no one wants to implement it. And before you ask, it's not trivial to solve it with the current implementation.

However, to run minisatip inside the same computer sharing an IP this is not a problem. So, you can do it. Futhermore, for the TVHeadend it's not a problem the current implementation. As I described before the "solution" is the use of the "X-SATIP-RTSP-Port" parameter. The current implementation of the minisatip includes it inside the HTTP Header of the "DESC.XML" file... check it:
https://github.com/catalinii/minisatip/blob/261069a4bc7daaaac768a66926bf75ac87bf77c8/src/satipc.c#L1696
And the TVHeadend reads it... see:
https://github.com/tvheadend/tvheadend/blob/f0dfae1bcfa7e26a07422a42b05c6e261a098579/src/input/mpegts/satip/satip.c#L1035

So I feel @Neolo has all tools to solve the problem. And if you want to use only the RTSP_over_TCP transport and use the loopback, then my suggestion is to protect your server using the firewall. Then the listening in the 0.0.0.0 "ANYADDR" will not be a problem.

Regards.

**Comment by Neolo** [comment]
Ok, thank you, I will use earlier suggested method of supplying customized xml configs from minisatip instances to TVH. I find it easier, without diving into the code much.

But still, implementing a bind-to-ip in general would make total sense here. Even TVH has this already, when acting as sat>ip server.

**Comment by lars18th** [comment]
Hi @Neolo ,

> But still, implementing a bind-to-ip in general would make total sense here. Even TVH has this already, when acting as sat>ip server.

Sure! That's an ancient request. However, it has low priority.

**Comment by Neolo** [comment]
I don't understand what -X path_to_custom_xml suppose to do. Whatever I supply doesn't change output from http://minisatip_ip/desc.xml
TVH does NOT take xml over file paths! And brining up a web server to serve it, is ridiculous.

**Comment by Neolo** [comment]
@lars18th 
Thanks for the patch https://github.com/catalinii/minisatip/pull/838, but it didn't make any difference, TVH doesn't see second instance, even with different names. It's either first or second minisatip when started, never at the same time.
![Screen Shot 2021-04-21 at 4 56 02 PM](https://user-images.githubusercontent.com/2836790/115649254-b4773b00-a2c2-11eb-9654-1f8ab73ca10c.png)

**Comment by 9000h** [comment]
You may have to change the mac address per minisatip

**Comment by lars18th** [comment]
Hi @Neolo ,

Please, try to do this:
- Identify your minisatip instances as "minisatip-A" and "minisatip-B". And not use the default RTSP 554 port, just use (as example) 1555 for A and 1556 for B. This is to guarantee that the client interprets correctly the alternative RTSP port (the use of the "X-SATIP-RTSP-Port" option). To set the name of the service use the `-I` parameter introduced in the PR #838.
- Then edit the corresponding `desc-a.xml` and `desc-b.xml` files as you know it.
- After execute _one_ instance only of minisatip, and start your tvheadend _without_ any internal tuner enabled and with the parameter `--satip_xml http://localhost:8081/desc.xml` (in this case the instance A is started with `-x 8081` too). Check if the SAT>IP tuner is available.
- Do the same with _one_ instance but configured for the instance B (different RTSP port and HTTP ports). Check if this is also visible or not.
- If both checks work, then start the tvheadend with the two instances active, and configured with two "--satip_xml" entries (like `--satip_xml http://localhost:8081/desc.xml --satip_xml http://localhost:8082/desc.xml`; remember to use "desc.xml" and not "-a" or "-b" !!!).
- If the last check fails, then edit the `desc-a.xml` and `desc-b.xml` files to use _different_ UUIDs.

I hope you can fix your problems.
Regards.

**Comment by lars18th** [comment]
> You may have to change the mac address per minisatip

I feel you want to say the UUID per minisatip... right? :stuck_out_tongue_winking_eye:

**Comment by Neolo** [comment]
> * Identify your minisatip instances as "minisatip-A" and "minisatip-B". And not use the default RTSP 554 port, just use (as example) 1555 for A and 1556 for B

Done

> * To set the name of the service use the `-I` parameter

Done

> * Then edit the corresponding `desc-a.xml` and `desc-b.xml` files as you know it

Done. Changed UUID, serial, representation IP.

> * After execute _one_ instance only of minisatip, and start your tvheadend _without_ any internal tuner enabled and with the parameter `--satip_xml http://localhost:8081/desc.xml` (in this case the instance A is started with `-x 8081` too). Check if the SAT>IP tuner is available.

Done, yes it's available "minisatip_a 192.168.0.57" as I named it. 192.168.0.57 is the default IP of the host, among another 4 ips.

> * Do the same with _one_ instance but configured for the instance B (different RTSP port and HTTP ports). Check if this is also visible or not.

Done, started second instance, both are running A and B.

> * If both checks work, then start the tvheadend with the two instances active, and configured with two "--satip_xml" entries (like `--satip_xml http://localhost:8081/desc.xml --satip_xml http://localhost:8082/desc.xml`; remember to use "desc.xml" and not "-a" or "-b" !!!).

Nope, only one is visible in TVH at a time and it's under same IP of 192.168.0.57, which is quite annoying.
I've tried to hardcode different ip, but I couldn't handle those woods of dependencies in the code.

**Comment by 9000h** [comment]
I mean -m 0040caa2c7ff, an there should no need to change the xml.

**Comment by Neolo** [comment]
Moreover, with custom XML used, both http://localhost:8081/desc.xml AND http://localhost:8082/desc.xml are reporting exact content, despite of customizations made! No wonder why TVH isn't able to show both instances.

**Comment by 9000h** [comment]
same UUID, try to change the mac with -m per instance.

**Comment by Neolo** [comment]
> same UUID, try to change the mac with -m per instance.

Will try other time, I've spent entire day on this issue with no success. Running second minisatip from different host makes TVH see both, local and remote minisatips, **just because they have different IPs**.

**Comment by 9000h** [comment]
satip servers need a uniq UUID and ID (-m and -D) that should do it if not probably the client is the issue

**Comment by Neolo** [comment]
> satip servers need a uniq UUID and ID (-m and -D) that should do it if not probably the client is the issue

Did you see what I just wrote in previous message? Simply running minisatip from different hosts makes it good, no UUID or other stuff modded.

**Comment by lars18th** [comment]
Hi @Neolo ,

> Did you see what I just wrote in previous message? Simply running minisatip from different hosts makes it good, no UUID or other stuff modded. Running it from same IP causes all kind of lap-overlap layered cake of headache.

Then this is a BUG inside TVHeadend. Perhaps it can't handle more than one SAT>IP server in the same IP address.
If this is the case, then you can do this to overcome the bug (after of course, write to the developer about this bug): create two alias IP addresses in your server, and use each one for each minisatip instances. Futhermore, read about how to create loopback aliases. Then you can use 127.0.0.2 and 127.0.0.3 just for your pourpose.

Finally it's confirmed that the problem is not related with the minisatip implementation.

**Comment by Neolo** [comment]
> Hi @Neolo ,
> 
> > Did you see what I just wrote in previous message? Simply running minisatip from different hosts makes it good, no UUID or other stuff modded. Running it from same IP causes all kind of lap-overlap layered cake of headache.
> 
> Then this is a BUG inside TVHeadend. Perhaps it can't handle more than one SAT>IP server in the same IP address.
> If this is the case, then you can do this to overcome the bug (after of course, write to the developer about this bug): create two alias IP addresses in your server, and use each one for each minisatip instances. Futhermore, read about how to create loopback aliases. Then you can use 127.0.0.2 and 127.0.0.3 just for your pourpose.
> 
> Finally it's confirmed that the problem is not related with the minisatip implementation.

You don't make any sense. Please pay more attention and read current page from the beginning. Tell me how you gonna "use each one for each minisatip instances"? Huh? This is the whole point of this issue opened - nohow.

**Comment by lars18th** [comment]
Hi @Neolo ,

I'm sorry if you can't resolve your problem. However, I need to say that I'm using more than one minisatip instances in the same server without any trouble. But in this case, I'm not using the autodiscovering and I configure the clients with the corresponding RTSP port. So it's nothing wrong with minisatip. Your problem is not related with the minisatip implementation in any sense (sorry if this sounds quite rude). If the TVHeadend can't handle more than one server in the same IP address then it's a problem of this tool.

Anyway, I need to make to comments:
1. I agree that minisatip requries to support a "binding address". This is a pending request #734.
2. Why you think that binding to an specific IP address will resolve your problem?

Regards.

**Comment by Neolo** [comment]
> 2\. Why you think that binding to an specific IP address will resolve your problem?

Because you are not reading posts. https://github.com/catalinii/minisatip/issues/836#issuecomment-824624848
Because separate IPs works with any clients. So instead of fixing client, better to fix a root problem of stop using 0.0.0.0

**Comment by 9000h** [comment]
sorry but you need a uniq UUID for each server the IP should not matter
https://tvheadend.org/issues/4176#note-6

**Comment by lars18th** [comment]
> > 2. Why you think that binding to an specific IP address will resolve your problem?
> 
> Because you are not reading posts. [#836 (comment)](https://github.com/catalinii/minisatip/issues/836#issuecomment-824624848)
> Because separate IPs works with any clients. So instead of fixing client, better to fix a root problem of stop using 0.0.0.0

Sorry @Neolo , but I feel you don't understand what we try to explain. Please note:
- The SAT>IP standard doesn't supports multiple SAT>IP servers sharing the same IP address. So one IP implies one server.
- Open source tools (like minisatip) supports multiple SAT>IP servers in the same IP address in different areas:
  - It can run using different RTSP port than the default 554.
  - The client mode doesn't requires the autodiscovering, and can use alternative RTSP ports.
  - As a server it supports the optional (not standard) label "X-SATIP-RTSP-Port".
  - It provides all the required functionality to not overlap RTP ports between different instances.

So, if you have some trouble in your environment, then try to fix the CLIENT part.

Futhermore, I can see how the IP binding to an specific IP address can fix your problem. If your think that you can fix your troubles using the default 554 port in single interfaces, then you're right. But this is a workaround for clients not supporting alternative RTSP ports. If you feel that this the definitive solution, then please open a PR providing your code. I feel the main developer (@catalinii ) is not interested on this. And personally, I've requested this in the past... and I feel it's good and profitable. However, I don't have the sufficient time to address this issue now.

I hope this helps you.
Regards.

**Comment by lars18th** [comment]
> sorry but you need a uniq UUID for each server the IP should not matter
> https://tvheadend.org/issues/4176#note-6

Yes, you're right. Different UUIDs it's mandatory. And time ago I've published a PR to print some data, including the UUID, in the UI as it's a very relevant information. However, the problem with the TVHeadend is that it doesn't support two SAT>IP servers in the same IP address. So what @Neolo wants to do it's to bind to an alias and use the default 554 RTSP port. For sure that's valid, but someone needs to update the code to bind to an specific IP address.

Regards.

**Comment by 9000h** [comment]
from what I understand he did not try to have a uniq UUID on the same IP

**Comment by Neolo** [comment]
> from what I understand he did not try to have a uniq UUID on the same IP

I couldn't try that, because option -X /my_custom_desc.xml, where I have altered the UUID was simply ignored, and it still spits out the default values at http://minisatip/desc.xml, which was read by TVH.
I will be happy to hardcode it, but where is it?

**Comment by 9000h** [comment]
the UUID is created from the mac address so do only use  a uniq  option -m 0040caa2c7ff for each server an do not use -X

**Comment by MusterGit** [comment]
> 
> 
> > from what I understand he did not try to have a uniq UUID on the same IP
> 
> I couldn't try that, because option -X /my_custom_desc.xml, where I have altered the UUID was simply ignored, and it still spits out the default values at http://minisatip/desc.xml, which was read by TVH.
> I will be happy to hardcode it, but where is it? Ot -m 028347923 (some random stuff) should change it?

Some weeks ago I wanted to use the custom xml too and was a bit surprised that the desc.xml remains default. I hope that this will be fixed some day, but for the meantime I found out that if you place your custom xml in the html folder you use it with tvh: 
`--satip_xml http://satip-server-dvbc:8080/custom_desc.xml`. The parameter `--satip_xml` can be used multiple times. Maybe this helps.

**Comment by Neolo** [comment]
> > > from what I understand he did not try to have a uniq UUID on the same IP
> > 
> > 
> > I couldn't try that, because option -X /my_custom_desc.xml, where I have altered the UUID was simply ignored, and it still spits out the default values at http://minisatip/desc.xml, which was read by TVH.
> > I will be happy to hardcode it, but where is it? Ot -m 028347923 (some random stuff) should change it?
> 
> Some weeks ago I wanted to use the custom xml too and was a bit surprised that the desc.xml remains default. I hope that this will be fixed some day, but for the meantime I found out that if you place your custom xml in the html folder you can use it with tvh:
> `--satip_xml http://satip-server-dvbc:8080/custom_desc.xml`. The parameter `--satip_xml` can be used multiple times.
> I'm using it with two servers:
> `--satip_xml http://satip-server-dvbs:8080/custom_desc.xml --satip_xml http://satip-server-dvbc:8080/custom_desc.xml`, maybe you could use `--satip_xml http://satip-server:8080/custom_desc.xml --satip_xml http://satip-server:8081/custom_desc.xml`
> I hope this helps.

I guess that should be a separate issue opened. What do you mean html folder? Folder of minisatip or what?

**Comment by MusterGit** [comment]
minisatip has a webserver integrated. The files for the webpage are inside a dedicated `html` folder (mostly inside /var, for me it's `/var/lib/minisatip/html`).

**Comment by lars18th** [comment]
> the UUID is created from the mac address so do only use a uniq option -m 0040caa2c7ff for each server an do not use -X

(@9000h ) Yes. You're right. And with my last PR #838 you can generate a different Service Name too.

> I hope this helps.

(@MusterGit )

This will not help only for one reason. If you use different RTSP port than the default 554, then it's required that the parameter "X-SATIP-RTSP-Port" will be included in the HTTP response containing the DESC.XML file. And the current minisatip code only does it when the request file is "\desc.xml".
Perhaps someone want to enhance this to support custom made XML description files. @catalinii ?

**Comment by Neolo** [comment]
> minisatip has a webserver integrated. The files for the webpage are inside a dedicated `html` folder (mostly inside /var, for me it's `/var/lib/minisatip/html`).

Hm, my folder is where I have compiled it. And how exactly I have to put custom xml file there?

**Comment by lars18th** [comment]
> Hm, my folder is where I have compiled it. And how exactly I have to put custom xml file there?

We're assuming that you put in inside the "/html" subfolder. It's the only valid place to put you custom files. If you see that any HTTP request returns the stock file, then it's because you don't put the file in the correct subdirectory. Try to copy any image (JPG or PNG) inside the folder, and wget it with the URL. It works? If not, then you're using incorrect configuration.

**Comment by Neolo** [comment]
> > Hm, my folder is where I have compiled it. And how exactly I have to put custom xml file there?
> 
> We're assuming that you put in inside the "/html" subfolder. It's the only valid place to put you custom files. If you see that any HTTP request returns the stock file, then it's because you don't put the file in the correct subdirectory. Try to copy any image (JPG or PNG) inside the folder, and wget it with the URL. It works? If not, then you're using incorrect configuration.

So put like what? Copy default output of desc.xml, alter and save in html folder as file? I have tried to serve custom xml to TVH via nginx. Guess what, nothing changed, TVH still receives all the default values somehow. Even if -A option placed.

**Comment by lars18th** [comment]
Hi @Neolo ,

Please, try to do steps one to one.
First, minisatip uses an INTERNAL http server. So **never** use nginx or any other server to provide info about the SAT>IP server. That's mandatory to exchange the "X-SATIP-RTSP-Port". If you don't use this parameter then you **can't** use another RTSP port than the default 554. Then it's impossible to use the same IP address.
Second, to check that you put the files in the correct folder, try to copy any file to it and download it when minisatip is running using the command `wget http://localhost:8081/testfile.png` (assuming you run the http server in the port 8081).
If you can download the file, then start to use custom DESC.XML files. My suggestion is to download the stock version with the wget and then edit it. Remember that if you use the parameter `--satip-xml` then you need to even use the URL http://localhost:8081/desc.xml`. Check until you see your edited file and not the stock autogenerated version.
After that, then start to execute the two instances, and check your clients. Regarding them I recommend to start with other clients different that the TVHeadend. Then you can test that both work as expected. And only after, you can try to configure the TVHeadend.

Regards.

Note: I've used the TVHeadend for years. But at time, I prefer to not use it at all. I use other alternatives... including the legendary Windows Media Center.

**Comment by Neolo** [comment]
> I've used the TVHeadend for years. But at time, I prefer to not use it at all. I use other alternatives... including the legendary Windows Media Center.

The reason I use TVH is because it's able to handle multi-frontend. https://github.com/catalinii/minisatip/issues/822

**Comment by Neolo** [comment]
Multiple instances of Minisatip in TVH are solved with just adding different -m MAC options to both, and using different names https://github.com/catalinii/minisatip/pull/838. No XML customizations done.

Step by step:
1) Merge the code from https://github.com/catalinii/minisatip/pull/838 and recompile
2) Configure minisatip instances as follow:
./minisatip -I minisatip -m 001122334410 -O -x 8081 -y 554 -D 1 ....
./minisatip -I minisatip -m 001122334420 -O -x 8082 -y 555 -D 2 ....
3) Start both and here is it:
![Screen Shot 2021-04-23 at 5 22 45 PM](https://user-images.githubusercontent.com/2836790/115945815-17023f80-a459-11eb-9a5f-ccde649b9067.png)

TIPS:
It's quite annoying when it all dumps log from both instances in syslog if systemd and -f used.
Use this template to run service and redirect log in separate file. Type must be FORKING, not SIMPLE. Modify and create different srvices per minisatip instance.

```
[Unit]
Description=Multi-threaded SAT>IP server
After=network-online.target
[Service]
User=root
Group=root
WorkingDirectory=/path/minisatip
Type=forking
ExecStart=/path/minisatip -I minisatip -m 001122334410 -O -x 8081 -y 554 -D 1 -F /var/log/minisatip.log ....
[Install]
WantedBy=multi-user.target


```

**Comment by 9000h** [comment]
@Neolo you are free to contribute and implement bind ip

**Comment by lars18th** [comment]
Perhaps some documentation of this topic can be added to the Wiki of the project: https://github.com/catalinii/minisatip/wiki

**Comment by Neolo** [comment]
> @Neolo you are free to contribute and implement bind ip

Tried already, too many dependencies in the code outside of my understanding, can't make it.

**Comment by lars18th** [comment]
> > @Neolo you are free to contribute and implement bind ip
> 
> Tried already, too many dependencies in the code outside of my understanding, can't make it.

Then try to write a wiki page... You can name it "How to run several instances of minisatip in the same server".

**Comment by Neolo** [comment]
It's all consolidated here https://github.com/catalinii/minisatip/issues/836#issuecomment-826026963
it's up to owners to merge it or not. My contribution, along with those who participated here, is done. But agin, it's only a workaround, doesn't mean it's solved.

**Comment by lars18th** [comment]
Hi @Neolo ,

> It's all consolidated here [#836 (comment)](https://github.com/catalinii/minisatip/issues/836#issuecomment-826026963)

Yes, and thank you for your contribution. However, at some point this issue will be closed. Then it's preferable to have this info documented in the Wiki. Therefore some other users will can read it in the future. See the number of topics included at time: https://github.com/catalinii/minisatip/wiki

> But again, it's only a workaround, doesn't mean it's solved.

I know it! Binding to an specific IP address it's a good idea. And tt's reported time ago. And we aggree with it. And it's an open issue. However, we need that some one wants to implement it.

Regards.

**Comment by 9000h** [comment]
thanks @lars18th for https://github.com/catalinii/minisatip/pull/838 all other is already well documented and easy to find

**Comment by ddimension** [comment]
Please notice my pull request #858 .

**Comment by ddimension** [comment]
> > Please notice my pull request #858 .
> 
> Thank you @ddimension !
> Why it doesn't work with 127.0.0.1 loopback address?

It works with loopback. But if you set the bind address and the bind interface also, you cannot reach minissdp. 
This is because internal traffic flows over loopback, which is not possible because if the interface binding. Just the usual linux behavior.

**Comment by lars18th** [comment]
> > Why it doesn't work with 127.0.0.1 loopback address?
> 
> It works with loopback. But if you set the bind address and the bind interface also, you cannot reach minissdp.
> This is because internal traffic flows over loopback, which is not possible because if the interface binding. Just the usual linux behavior.

Ok. Then why not just check the use of both options at the same time and write a warning message then? You think that's a good idea?
In any case I hope that @catalinii will want to merge this PR.

Good job @ddimension ! :+1:

**Comment by Jalle19** [comment]
AFAICT this was closed by https://github.com/catalinii/minisatip/pull/858
